corpusreaders/src/main/java/edu/illinois/cs/cogcomp/nlp/corpusreaders/ontonotes/OntonotesNamedEntityReader.java

```diff
@@ -0,0 +1,247 @@
+ * http://cogcomp.cs.illinois.edu/
+ */
+package edu.illinois.cs.cogcomp.nlp.corpusreaders.ontonotes;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import edu.illinois.cs.cogcomp.annotation.AnnotatorException;
+import edu.illinois.cs.cogcomp.annotation.XmlTextAnnotationMaker;
+import edu.illinois.cs.cogcomp.core.datastructures.IntPair;
+import edu.illinois.cs.cogcomp.core.datastructures.ViewNames;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.Constituent;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.SpanLabelView;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.TextAnnotation;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.View;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.XmlTextAnnotation;
+import edu.illinois.cs.cogcomp.core.io.LineIO;
+import edu.illinois.cs.cogcomp.core.utilities.SerializationHelper;
+import edu.illinois.cs.cogcomp.core.utilities.XmlDocumentProcessor;
+import edu.illinois.cs.cogcomp.core.utilities.XmlDocumentProcessor.SpanInfo;
+import edu.illinois.cs.cogcomp.nlp.corpusreaders.AnnotationReader;
+import edu.illinois.cs.cogcomp.nlp.corpusreaders.CorpusReaderConfigurator;
+import edu.illinois.cs.cogcomp.nlp.corpusreaders.ereReader.CoNLL2002Writer;
+import edu.illinois.cs.cogcomp.nlp.corpusreaders.ontonotes.utils.DocumentIterator;
+import edu.illinois.cs.cogcomp.nlp.tokenizer.StatefulTokenizer;
+import edu.illinois.cs.cogcomp.nlp.utility.TokenizerTextAnnotationBuilder;
+
+/**
+ * This class will traverse a directory hierarchy searching for any file with a 
+ * certain file extension. The file is expected to be a penn treebank parse file in
+ * text format. It will go any number of levels deep. The iteration produces a
+ * TextAnnotation for each file.
+ * @author redman
+ */
+public class OntonotesNamedEntityReader extends AnnotationReader<XmlTextAnnotation> {
+    
+    /** tags containing text. */
+    static final private Set<String> tagsWithText = new HashSet<>();
+    
+    /** tags we can just ignore. */
+    static final private Set<String> dropTags = new HashSet<>();
+
+    /** define the attributes we want to keep for the tags we have. */
+    static final private Map<String, Set<String>> tagsWithAtts = new HashMap<>();
+    static {
+        Set<String> docAttrs = new HashSet<>();
+        docAttrs.add("docno");
+        tagsWithAtts.put("doc", docAttrs);
+        Set<String> nameAttrs = new HashSet<>();
+        nameAttrs.add("type");
+        tagsWithAtts.put("enamex", nameAttrs);
+    }
+
+    /** the home directory to traverse. */
+    protected final String homeDirectory;
+    
+    /** the name of the resulting view. */
+    protected String parseViewName = "OntonotesNER";
```

Comment:
IF the view is created by a reader from a corpus, it's better to call it NER_ONTONOTES_GOLD. We use constants defined in a single class so that all applications use a consistent label -- avoids errors due to typos etc.